static/app/components/modals/debugFileCustomRepository/http.tsx

```diff
@@ -0,0 +1,285 @@
+                ...data,
+                username: e.target.value,
+              })
+            }
+          />
+        </Field>
+        <Field
+          label={t('Password')}
+          inline={false}
+          help={t('Password for HTTP basic auth')}
+          flexibleControlStateSize
+          stacked
+        >
+          <PasswordInput
+            type={data.password === undefined ? 'text' : 'password'}
+            name="url"
+            placeholder={
+              data.password === undefined ? t('(Password unchanged)') : 'open-sesame'
+            }
+            value={data.password}
+            onChange={e =>
+              setData({
+                ...data,
+                password: e.target.value,
+              })
+            }
+          />
+          {(data.password === undefined ||
+            (typeof data.password === 'string' && !!data.password)) && (
+            <ClearPasswordButton
+              onClick={handleClearPassword}
+              icon={<IconClose size="14px" />}
+              size="xsmall"
+              title={t('Clear password')}
+              label={t('Clear password')}
+              borderless
+            />
+          )}
+        </Field>
+        <Divider />
+        <StyledSelectField
+          name="layout.type"
+          label={t('Directory Layout')}
+          help={t('The layout of the folder structure.')}
+          choices={Object.keys(DEBUG_SOURCE_LAYOUTS).map(key => [
+            key,
+            DEBUG_SOURCE_LAYOUTS[key],
+          ])}
+          value={data['layout.type']}
+          onChange={value =>
+            setData({
+              ...data,
+              ['layout.type']: value,
+            })
+          }
+          inline={false}
+          flexibleControlStateSize
+          stacked
+        />
+        <StyledSelectField
+          name="layout.casing"
+          label={t('Path Casing')}
+          help={t('The case of files and folders.')}
+          choices={Object.keys(DEBUG_SOURCE_CASINGS).map(key => [
+            key,
+            DEBUG_SOURCE_CASINGS[key],
+          ])}
+          value={data['layout.casing']}
+          onChange={value =>
+            setData({
+              ...data,
+              ['layout.casing']: value,
+            })
+          }
+          inline={false}
+          flexibleControlStateSize
+          stacked
+        />
+      </Body>
+      <Footer>
+        <Button
+          onClick={handleSubmit}
+          priority="primary"
+          disabled={isFormInvalid() || formUnchanged()}
+        >
+          {t('Save changes')}
+        </Button>
+      </Footer>
+    </Fragment>
+  );
+}
+
+export default Http;
+
+const StyledSelectField = styled(SelectField)`
+  padding-right: 0;
+`;
+
+const Divider = styled('div')`
+  height: 1px;
+  background-color: ${p => p.theme.gray200};
+  margin-bottom: ${space(2)};
+`;
+
+const PasswordInput = styled(Input)`
+  /* 10px of the input default padding-right + 22px of button size */
```

Comment:
Can we extract this into code?

Reply:
not sure what you mean here

Reply:
e.g. extract `22` into a constant, and calculate PasswordInput's padding based on that. Just in case someone comes in to change the icon size/padding.

Reply:
I've updated the code. pls lmk if it's fine :)